ui/search/nuxeo-search-form-layout.js

```diff
@@ -0,0 +1,154 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import '@polymer/paper-button/paper-button.js';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+import '@polymer/paper-input/paper-input.js';
+import '@polymer/paper-spinner/paper-spinner-lite.js';
+import '@polymer/paper-toggle-button/paper-toggle-button.js';
+import '@nuxeo/nuxeo-elements/nuxeo-page-provider.js';
+import '@nuxeo/nuxeo-elements/nuxeo-search.js';
+import '../nuxeo-layout.js';
+import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
+import { RoutingBehavior } from '../nuxeo-routing-behavior.js';
+import '../nuxeo-data-list/nuxeo-data-list.js';
+import '../nuxeo-document-permissions/nuxeo-document-permissions.js';
+import '../widgets/nuxeo-dialog.js';
+import '../widgets/nuxeo-select.js';
+import '../nuxeo-document-thumbnail/nuxeo-document-thumbnail.js';
+
+{
+  /**
+  `nuxeo-search-form-layout`
+  @group Nuxeo UI
+  @element nuxeo-search-form-layout
+  */
+  class SearchFormLayout extends mixinBehaviors([I18nBehavior, RoutingBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <nuxeo-layout
+          id="layout"
+          href="[[_formHref(provider, searchName)]]"
+          model="[[_formModel(provider, aggregations, params)]]"
+          error="[[i18n('documentSearchForm.layoutNotFound', searchName)]]"
+          on-element-changed="_formChanged"
+        ></nuxeo-layout>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-search-form-layout';
+    }
+
+    static get importMeta() {
+      return import.meta;
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The `nuxeo-page-provider` instance used to perform the search.
+         * */
+        provider: String,
+        /**
+         * The name of the search layout.
+         * */
+        searchName: String,
+        /**
+         * The parameters passed on to `provider`.
+         * */
+        params: {
+          type: Object,
+          notify: true,
+        },
+        skipAggregates: {
```

Comment:
Missing documentation here. We can copy the one from nuxeo-results-view, for example.